presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushProjectionThroughJoin.java

```diff
@@ -57,6 +58,10 @@
         PlanNode leftChild = joinNode.getLeft();
         PlanNode rightChild = joinNode.getRight();
 
+        if (joinNode.getType() != INNER) {
```

Comment:
Why is this an issue? It should be possible (although maybe not productive) to push a projection through the left side of a left join, for instance.

Reply:
> Why is this an issue? It should be possible (although maybe not productive)

This was an issue because there was a bug in `JoinGraph.Builder#visitGroupReference` which built wrong `symbol -> node` map when projection was pushed through outer join.

However, I think the key here is that pushing projections through outer join is rather not productive (e.g. projected rows might have been filtered out) and not really needed for cross join enumeration/join reordering (so why do the pushdown?)